### PR TITLE
Java: improve naming and description of SqlUnescaped.ql

### PR DESCRIPTION
**Pull request by smowton**
Since the main thing it's objecting to is concatenation not lack of escaping (in particular it doesn't look for escaping sanitizers), rename and re-describe it accordingly.

### CODE REVIEW

**Comment by atorralba** [comment]
Looks good to me, although we should also reword this other mention of "neutralizing special characters":

https://github.com/github/codeql/pull/11730/files#diff-e5a9750a680159ef922f625283c8b8e869c11b1629e2d58b09ce1a300b7cff38R8

Suggestion:
```diff
Even when the components of a SQL query are not fully controlled by
-a user, it is a vulnerability to concatenate those components into a
-SQL query without neutralizing special characters.
+a user, it is a vulnerability to build the query by directly
+concatenating those components.
```

**Comment by smowton** [comment]
@atorralba done

**Comment by atorralba** [comment]
LGTM, aside from this needing a change note, and that apparently existing alerts will be closed and reopened massively because of the alert message change.

**Review by egregius313 — DISMISSED** [review]
LGTM

File: java/ql/src/change-notes/2023-01-03-renamed-sql-unescaped.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* The name, description and alert message for the query `java/concatenated-sql-query` have been altered to emphasise that the query flags the use of string concatenation to construct SQL queries, not the lack of appropriate escaping. The query's files have been renamed from `SqlUnescaped.ql` and `SqlUnescapedLib.qll` to `SqlConcatenated.ql` and `SqlConcatenatedLib.qll` respectively; in the unlikely event your custom configuration or queries refer to either of these files by name, those references will need to be adjusted.
```
Comment (adityasharad) [inline java/ql/src/change-notes/2023-01-03-renamed-sql-unescaped.md:4]: Perhaps mention that the query ID has not changed, so alerts for this query will not be affected.

**Review by adityasharad — APPROVED** [review]

**Review by atorralba — APPROVED** [review]

**Pull request merged**